File: proof_of_work/pow.py
#!/bin/python

import sys
import time
import hashlib
from struct import unpack, pack

timestamp = str(time.time()) 		# Work timestamp
message = "This is a random message." 	# Plaintext message
nonce = 0				# 
guess = 99999999999999999999
payload = timestamp + message
throttle = 100000000
target = 2**64 / throttle

#encoding for windows
payloadHash = hashlib.sha256(payload.encode('utf-8')).digest() 

start = time.time()
while guess > target:
	nonce += 1
	guess, = unpack('>Q',hashlib.sha256(hashlib.sha256(pack('>Q',nonce) + payloadHash.encode('utf-8')).digest()).digest()[0:8])
	print(guess);
end = time.time()

print "%s:%s:%s:%s:%s:%s:%s" % (timestamp, message, nonce, guess, payload, target, end-start)

File: proof_of_work/pow.js
var crypto = require("crypto");
var hash = crypto.createHash("sha256");

var timestamp = Date.now();
var message = "Random Message";
var nonce = 0;
guess = 99999999999999999999;
payload = timestamp + message;

throttle = 1000000;
target = Math.pow(2, 64) / throttle;

var start = Date.now();


while (guess > target) {
    hashA = crypto.createHash("sha256");
    hashB = crypto.createHash("sha256");
    nonce+=1;
    guess = parseInt(
              hashB.update(
                hashA.update(payload + nonce).digest('hex'))
              .digest('hex') 
            .substring(0,16), 16);
    // Turn on to debug
    //guess = hash.update(payload + nonce).digest('hex');

    // 16 Hex nibbles
    console.log(guess)
    //console.log(guess.substring(0,16));
    // 16 x 4 bits = 64 bits in decimal
    //console.log(parseInt(guess.substring(0,16), 16));
    //console.log(target)
}
var end = Date.now()
console.log("Run time: ", (end-start)/1000);